### ISSUE TITLE
Metadata search does not support S3 signing changes in ECS 3.0

### ISSUE THREAD

**Issue by brokenjacobs** [djannot/ecsui#4]
Attempting to search a bucket with metadata search enabled returns the following error:
<Error><Code>SignatureDoesNotMatch</Code><Message>The request signature we calculated does not match the signature you provided. Check your Secret Access Key and signing method. For more information, see REST Authentication and SOAP Authentication for details.</Message><RequestId>ac1f2287:159465e69f6:5e66:7</RequestId></Error>

My understanding is the signing requirement changed in 3.0 ECS, which would explain this error.

**Comment by djannot** [comment]
Yes, I've updated the code to make it compatible with ECS 3.0 as I don't thin many people are using metadata search with an older version. If you don't plan to upgrade soon, I can add an environment variable to use the old signature algorithm.

**Comment by brokenjacobs** [comment]
I was having this issue with ECS 3.0. When was the change made?

**Comment by brokenjacobs** [comment]
![screen shot 2017-02-15 at 2 45 30 pm](https://cloud.githubusercontent.com/assets/3159463/22996848/6c9bda04-f38d-11e6-9693-693b2032c947.png)

This is against ECS 3.0.0 HF1.

**Comment by djannot** [comment]
@brokenjacobs The modification has been implemented in August 2016.

Are you using the Docker container or are you compiling the code yourself ?
What information are you indicating in the login page ?
On which action do you have this error ?

**Comment by brokenjacobs** [comment]
I was using the docker container. I am able to search using the spark connector, but not ensuing.

-- 
Steve Jacobs
Sent with Airmail

On February 18, 2017 at 1:48:45 PM, Denis Jannot (notifications@github.com) wrote:

@brokenjacobs The modification has been implemented in August 2016.

Are you using the Docker container or are you compiling the code yourself ?
What information are you indicating in the login page ?
On which action do you have this error ?

**Comment by djannot** [comment]
@brokenjacobs Can you please tell me the syntax you use for the query ?
I'll reproduce and fix it.

**Comment by brokenjacobs** [comment]
Roger that, sorry for the late response I haven't had time to check this.
